swap/src/network/spot_price.rs

```diff
@@ -41,7 +41,28 @@ pub struct Request {
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct Response {
-    pub xmr: monero::Amount,
+    pub xmr: Option<monero::Amount>,
+    pub error: Option<Error>,
+}
+
+#[derive(Clone, Debug, thiserror::Error, Serialize, Deserialize)]
```

Comment:
Deriving `Error` on a struct that is part of the wire protocol is kind of weird, can we separate this please so that we have a clearly defined, external API?

Reply:
All these errors are related to the `spot_price` protocol that's why I moved them into the protocol. I think the dedicated `NetworkBehaviour` would solve all the problems that came up (then the Error could also be defined there). I will see to work on that.

Reply:
I agree with all you said. What I am saying is: we may want to have separate structs for what is _serialized_ vs what is interally used to pass around errors. Usually, you'd only implement `std::error::Error` for things that go into a `Result` and where you use `?`.